### ISSUE TITLE
st_rasterize - template and align=TRUE ?

### ISSUE THREAD

**Issue by JimShady** [r-spatial/stars#489]
Hi Stars team,

Quick question about using [st_rasterize](https://r-spatial.github.io/stars/reference/st_rasterize.html). I am using it like so:

```
st_rasterize(sf_polygons,
                  template = read_stars("template.tif"),
                  file = "output_raster.tif",
                  driver = "GTiff")
```

However my sf_polygons object covers the whole world, whereas my template.tif only covers the UK. So the result raster (output_raster.tif) also only covers the UK.

I was wondering/hoping that there is some sort of 'align=TRUE' option, similar to in the {terra} package:

https://rdrr.io/github/rspatial/terra/man/project.html

So that the template is just used for resolution and alignment, rather than restricting the output area to the size of the template. I hope I'm explaining myself well enough.

I note that it's possible to pass additional gdalrasterize arguments to the function, and suspect that I might be able to achieve my desired output by using these commands, but I'm unsure how to achieve that exactly.

https://gdal.org/programs/gdal_rasterize.html#cmdoption-gdal_rasterize-te

Thanks, Jim

**Comment by edzer** [comment]
Could you please try out whether this works? I only used very artificial examples, not all the way into `st_rasterize()`.

**Comment by JimShady** [comment]
Awesome, thanks for trying to implement. I'll give it a go on Monday and feedback here.

**Comment by JimShady** [comment]
Hi @edzer  . This seems to be working - thank you! I am getting an error .... but I think it's probably unrelated.

```
st_rasterize(sf_polygons,
                  template = read_stars("template.tif"),
                  file = "output_raster.tif",
                  driver = "GTiff",
                  align=TRUE)
```


Error : cannot allocate vector of size 6035.8 Gb

Not quite sure why I'm getting this, as the input sf_polygons is integers between 0 and 100, and the template.tif is of type INT1U (8-bit unsigned). So even though my raster is worldwide and quite high resolution (0.0002 degrees), I was expecting something more in the 20-30GB range. Most of the raster that I create will be NA values. Then the rest between 0 and 100. So I thought it would be manageable.

If you think that this is unrelated to align=TRUE please feel free to close the ticket. Thanks again.

**Comment by edzer** [comment]
Maybe you can retry this, with argument `proxy = TRUE` ?

**Comment by JimShady** [comment]
Thanks, will try that now.

To make sure that the output is a 8 bit unsigned raster, I think I can do this right?

```
st_rasterize(sf_polygons,
                  template = read_stars("template.tif"),
                  file = "output_raster.tif",
                  driver = "GTiff",
                  proxy=TRUE,
                  options = "-ot BYTE")
```

**Comment by JimShady** [comment]
So this runs .... but the returned raster is only the coverage of the input template, rather than the whole world.

**Comment by edzer** [comment]
A reprex, for instance with a built-in dataset (see `?read_stars`) would help!

**Comment by JimShady** [comment]
Never made a reprex before so not quite sure how. I'll read about it now and try.

As I'd just finished making a ZIP of the shapefile and the template.tif anyway, here it is if you want it in the meantime.

https://drive.google.com/file/d/16H3aDEvcgRk65TgU80pFqJVzrYq8hjmM/view?usp=sharing

**Comment by JimShady** [comment]
Hi @edzer . I guess this isn't a 'true' reprex, but hopefully is easy to run and test?

```
library("sf")
library("stars")

input <- st_read("https://github.com/JimShady/nffc_travel_vis/raw/main/flopros.gpkg")

download.file("https://github.com/JimShady/nffc_travel_vis/raw/main/template.zip",  "template.zip")

unzip('template.zip', exdir = 'downloaded')

template <- read_stars("downloaded/template.tif")

stars::st_rasterize(input, template, align = TRUE, proxy = TRUE, file = 'output.tif')
```

**Comment by edzer** [comment]
Thanks; when trying to write the template to disk (with `gdal_create`), I'm getting a GDAL error
```
ERROR 3: NA: Free disk space available is 420535185408 bytes, whereas 586869486252 are at least necessary.
```
which sounds a bit like  you may want to reduce your problem size-wise.

**Comment by JimShady** [comment]
Thanks. I initially believed that this was an error, but I now realise it's literally that the data I'm trying to create is too big. I will need to do my work country by country basis I think, rather than trying to attempt to do it globally all at once.
